### PR TITLE
Release GIL in prepare_actor_checkpoint

### PR DESCRIPTION
**Pull request by raulchen**

### CODE REVIEW

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -352,8 +352,12 @@ cdef class RayletClient:
 
     def prepare_actor_checkpoint(self, ActorID actor_id):
         cdef CActorCheckpointID checkpoint_id
-        check_status(self.client.get().PrepareActorCheckpoint(
-            actor_id.data, checkpoint_id))
+        cdef CActorID c_actor_id = actor_id.data
+        # PrepareActorCheckpoint will wait for raylet's reply, release
+        # the GIL so other Python threads can run.
+        with nogil:
+            check_status(self.client.get().PrepareActorCheckpoint(
```
Comment (robertnishihara) [inline python/ray/_raylet.pyx]: Is `PrepareActorCheckpoint` a blocking call, which waits for a response from the raylet? If yes, then we definitely need to release the GIL. If not, then we could still release the GIL, though in that case there are many other methods like this where we should release the GIL.

Comment (robertnishihara) [reply python/ray/_raylet.pyx]: Ok, I just checked and see that it waits for a response.

**Review by robertnishihara — APPROVED** [review]

File: python/ray/_raylet.pyx [hunk 2]
```diff
@@ -352,8 +352,12 @@ cdef class RayletClient:
 
     def prepare_actor_checkpoint(self, ActorID actor_id):
         cdef CActorCheckpointID checkpoint_id
-        check_status(self.client.get().PrepareActorCheckpoint(
-            actor_id.data, checkpoint_id))
+        cdef CActorID c_actor_id = actor_id.data
```
Comment (robertnishihara) [inline python/ray/_raylet.pyx]: Btw, I don't think it is necessary to do this is it? Because `ActorID` should already be a C++ object.

Comment (raulchen) [reply python/ray/_raylet.pyx]: I'm not sure about this. I thought `ActorID` was a Python object.

**Pull request merged**